web-console/src/dialogs/status-dialog/status-dialog.spec.tsx

```diff
@@ -27,4 +27,19 @@ describe('status dialog', () => {
     render(statusDialog);
     expect(document.body.lastChild).toMatchSnapshot();
   });
+  it('filters data that contains input', () => {
+    const data = [
+      'org.apache.druid.common.gcp.GcpModule',
+      'org.apache.druid.common.aws.AWSModule',
+      'io.imply.druid.UtilityBeltModule',
+    ];
+    let filterFunction;
+    filterFunction = (filter: { id: string | number; value: string }, row: { [x: string]: any }) =>
+      String(row[filter.id]).includes(filter.value);
```

Comment:
This method looks like it's testing a clone of the logic in the component. It's better if you can test the component via its interfaces/properties instead of testing a clone of the component's internal logic. That way, if the internal logic of the component changes later, your test will still be validating the component's behavior (whereas the clone will be stale).

Reply:
I wasn't sure how to test an instance of the component directly, so I followed what Vadim said about testing the function itself.

Reply:
If testing the component directly is not possible, then perhaps extracting the method to a library is an option? The library method can have a unit test and then the component can use that same library method.

Reply:
I've made the changes so it tests the component's filtering directly.